src/plugins/intel_gpu/tests/functional/behavior/model_cache.cpp

```diff
@@ -0,0 +1,210 @@
+//
+
+#include <cstdio>
+
+#include "base/ov_behavior_test_utils.hpp"
+#include "common_test_utils/common_utils.hpp"
+#include "common_test_utils/file_utils.hpp"
+#include "common_test_utils/ov_tensor_utils.hpp"
+#include "common_test_utils/subgraph_builders/2_input_subtract.hpp"
+#include "common_test_utils/subgraph_builders/concat_with_params.hpp"
+#include "common_test_utils/subgraph_builders/conv_bias.hpp"
+#include "common_test_utils/subgraph_builders/conv_pool_relu.hpp"
+#include "common_test_utils/subgraph_builders/conv_pool_relu_no_reshapes.hpp"
+#include "common_test_utils/subgraph_builders/conv_pool_relu_non_zero.hpp"
+#include "common_test_utils/subgraph_builders/convert_transpose.hpp"
+#include "common_test_utils/subgraph_builders/detection_output.hpp"
+#include "common_test_utils/subgraph_builders/kso_func.hpp"
+#include "common_test_utils/subgraph_builders/matmul_bias.hpp"
+#include "common_test_utils/subgraph_builders/multi_single_conv.hpp"
+#include "common_test_utils/subgraph_builders/multiple_input_outpput_double_concat.hpp"
+#include "common_test_utils/subgraph_builders/nested_branch_conv_concat.hpp"
+#include "common_test_utils/subgraph_builders/nested_split_conv_concat.hpp"
+#include "common_test_utils/subgraph_builders/read_concat_split_assign.hpp"
+#include "common_test_utils/subgraph_builders/single_concat_with_constant.hpp"
+#include "common_test_utils/subgraph_builders/single_conv.hpp"
+#include "common_test_utils/subgraph_builders/single_split.hpp"
+#include "common_test_utils/subgraph_builders/split_concat.hpp"
+#include "common_test_utils/subgraph_builders/split_conv_concat.hpp"
+#include "common_test_utils/subgraph_builders/split_multi_conv_concat.hpp"
+#include "common_test_utils/subgraph_builders/ti_with_lstm_cell.hpp"
+#include "common_test_utils/test_common.hpp"
+#include "openvino/pass/serialize.hpp"
+
+namespace {
+class CheckWeightlessCacheAccuracy : public ::testing::Test {
+protected:
+    std::shared_ptr<ov::Model> model;
+    std::string xml_path;
+    std::string bin_path;
+    std::string cache_path;
+
+    void SetUp() override;
+    void TearDown() override;
+    void run();
+};
+
+void CheckWeightlessCacheAccuracy::SetUp() {
+    std::string filePrefix = ov::test::utils::generateTestFilePrefix();
+    xml_path = filePrefix + ".xml";
+    bin_path = filePrefix + ".bin";
+    cache_path = filePrefix + ".blob";
+}
+
+void CheckWeightlessCacheAccuracy::TearDown() {
+    std::remove(xml_path.c_str());
+    std::remove(bin_path.c_str());
+    std::remove(cache_path.c_str());
+}
+
+void CheckWeightlessCacheAccuracy::run() {
+    ov::AnyMap config = {{"CACHE_MODE", ov::CacheMode::OPTIMIZE_SIZE}, {"WEIGHTS_PATH", bin_path}};
+    auto core = ov::test::utils::PluginCache::get().core();
+    ov::pass::Serialize(xml_path, bin_path).run_on_model(model);
+
+    ov::CompiledModel compiled_model;
+    OV_ASSERT_NO_THROW(compiled_model = core->compile_model(xml_path, ov::test::utils::DEVICE_GPU, config));
```

Comment:
I wonder if Core can automatically set weights path when model is loaded from IR? Would be nice if users won't need to set this property manually all the time

Reply:
It was like that in an earlier version but I changed it after it was requested that it should only exist in runtime: https://github.com/openvinotoolkit/openvino/pull/25731#discussion_r1799301914.

Reply:
My suggestion is a little bit different. I agree that weighs path shouldn't be a part of serialized IR. But we may do something like this
```
ov::Model model = core.read_model(...); // it saves weights path as non-serializable attribute in ov::Model
ov::CompiledModel = core.compile_model(model); // internally core converts this attribute of ov::Model to ov::weights_path property and pass it to the plugin
```

In that case users can avoid setting weights path explicitly and it will be mainly needed if they call `import_model` explicitly

Reply:
Sure, that sounds good. What about saving it in the cache file? I would assume it shouldn't be exported to cache for the same reason it shouldn't be serialized in the IR?

Reply:
Yes, I think it shouldn't be exported to the cache too.

Reply:
> ov::Model model = core.read_model(...); // it saves weights path as non-serializable attribute in ov::Model

I contacted the core/serialization team and received information that currently all model attributes are serialized and it's impossible to have a non-serializable one without modifying the serialization mechanism itself. I feel that such change is outside of scope of this PR. What are your thoughts on this?